## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 450086`
Commit: [`16638cbed3469cb71aacff8948048e52d20ed8e7`](https://github.com/NixOS/nixpkgs/commit/16638cbed3469cb71aacff8948048e52d20ed8e7) ([subsequent changes](https://github.com/NixOS/nixpkgs/compare/16638cbed3469cb71aacff8948048e52d20ed8e7..pull/450086/head))
Merge: [`d1353cfc93bc19ab490a68daedb1dd96826020bb`](https://github.com/NixOS/nixpkgs/commit/d1353cfc93bc19ab490a68daedb1dd96826020bb)

Logs: https://github.com/liberodark/nixpkgs-review-gha/actions/runs/19759956761


---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>mklittlefs</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>mklittlefs</li>
  </ul>
</details>

---
### `x86_64-darwin` (sandbox = relaxed)
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>mklittlefs</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-darwin`</summary>
<details>
<summary>mklittlefs</summary>
<pre>patching script interpreter paths in tests/test_create
tests/test_create: interpreter directive changed from &quot;#!/bin/bash&quot; to &quot;/nix/store/lbkni2ik42yc5hic017z8r6h5gn1gy04-bash-5.3p3/bin/bash&quot;
Running phase: configurePhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;configurePhase&quot; }
no configure script, doing nothing
Running phase: buildPhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;buildPhase&quot; }
build flags: -j3 SHELL=/nix/store/lbkni2ik42yc5hic017z8r6h5gn1gy04-bash-5.3p3/bin/bash BUILD_CONFIG_NAME=-nixos CPPFLAGS=-DLFS_NAME_MAX=255
clang++ -std=gnu++11 -Os -Wall -Wextra -Werror   -Itclap -Iinclude -Ilittlefs -I. -D VERSION=\&quot;4.1.0\&quot; -D LITTLEFS_VERSION=\&quot;2.11.1\&quot; -D BUILD_CONFIG=\&quot;-DLFS_NAME_MAX=255\&quot; -D BUILD_CONFIG_NAME=\&quot;-nixos\&quot; -D __NO_INLINE__ -D LFS_NAME_MAX=255 -DLFS_NAME_MAX=255  -c -o main.o main.cpp
clang -std=gnu99 -Os -Wall -Wextra -Werror   -Itclap -Iinclude -Ilittlefs -I. -D VERSION=\&quot;4.1.0\&quot; -D LITTLEFS_VERSION=\&quot;2.11.1\&quot; -D BUILD_CONFIG=\&quot;-DLFS_NAME_MAX=255\&quot; -D BUILD_CONFIG_NAME=\&quot;-nixos\&quot; -D __NO_INLINE__ -D LFS_NAME_MAX=255 -DLFS_NAME_MAX=255  -c -o littlefs/lfs.o littlefs/lfs.c
clang -std=gnu99 -Os -Wall -Wextra -Werror   -Itclap -Iinclude -Ilittlefs -I. -D VERSION=\&quot;4.1.0\&quot; -D LITTLEFS_VERSION=\&quot;2.11.1\&quot; -D BUILD_CONFIG=\&quot;-DLFS_NAME_MAX=255\&quot; -D BUILD_CONFIG_NAME=\&quot;-nixos\&quot; -D __NO_INLINE__ -D LFS_NAME_MAX=255 -DLFS_NAME_MAX=255  -c -o littlefs/lfs_util.o littlefs/lfs_util.c
main.cpp:531:20: error: variable length arrays in C++ are a Clang extension [-Werror,-Wvla-cxx-extension]
  531 |     uint8_t buffer[littlefsFile-&gt;size];
      |                    ^~~~~~~~~~~~~~~~~~
main.cpp:531:20: note: function parameter &#x27;littlefsFile&#x27; with unknown value cannot be used in a constant expression
main.cpp:530:47: note: declared here
  530 | bool unpackFile(const char *lfsDir, lfs_info *littlefsFile, const char *destPath) {
      |                                               ^
1 error generated.
make: *** [&lt;builtin&gt;: main.o] Error 1</pre>
</details>
</details>

---
### `aarch64-darwin` (sandbox = relaxed)
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>mklittlefs</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `aarch64-darwin`</summary>
<details>
<summary>mklittlefs</summary>
<pre>Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;updateAutotoolsGnuConfigScriptsPhase&quot; }
Running phase: configurePhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;configurePhase&quot; }
no configure script, doing nothing
Running phase: buildPhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;buildPhase&quot; }
build flags: -j3 SHELL=/nix/store/p0k9r5h8qs7220xdbdihhfgzwjcly70x-bash-5.3p3/bin/bash BUILD_CONFIG_NAME=-nixos CPPFLAGS=-DLFS_NAME_MAX=255
clang++ -std=gnu++11 -Os -Wall -Wextra -Werror   -Itclap -Iinclude -Ilittlefs -I. -D VERSION=\&quot;4.1.0\&quot; -D LITTLEFS_VERSION=\&quot;2.11.1\&quot; -D BUILD_CONFIG=\&quot;-DLFS_NAME_MAX=255\&quot; -D BUILD_CONFIG_NAME=\&quot;-nixos\&quot; -D __NO_INLINE__ -D LFS_NAME_MAX=255 -DLFS_NAME_MAX=255  -c -o main.o main.cpp
clang -std=gnu99 -Os -Wall -Wextra -Werror   -Itclap -Iinclude -Ilittlefs -I. -D VERSION=\&quot;4.1.0\&quot; -D LITTLEFS_VERSION=\&quot;2.11.1\&quot; -D BUILD_CONFIG=\&quot;-DLFS_NAME_MAX=255\&quot; -D BUILD_CONFIG_NAME=\&quot;-nixos\&quot; -D __NO_INLINE__ -D LFS_NAME_MAX=255 -DLFS_NAME_MAX=255  -c -o littlefs/lfs.o littlefs/lfs.c
clang -std=gnu99 -Os -Wall -Wextra -Werror   -Itclap -Iinclude -Ilittlefs -I. -D VERSION=\&quot;4.1.0\&quot; -D LITTLEFS_VERSION=\&quot;2.11.1\&quot; -D BUILD_CONFIG=\&quot;-DLFS_NAME_MAX=255\&quot; -D BUILD_CONFIG_NAME=\&quot;-nixos\&quot; -D __NO_INLINE__ -D LFS_NAME_MAX=255 -DLFS_NAME_MAX=255  -c -o littlefs/lfs_util.o littlefs/lfs_util.c
main.cpp:531:20: error: variable length arrays in C++ are a Clang extension [-Werror,-Wvla-cxx-extension]
  531 |     uint8_t buffer[littlefsFile-&gt;size];
      |                    ^~~~~~~~~~~~~~~~~~
main.cpp:531:20: note: function parameter &#x27;littlefsFile&#x27; with unknown value cannot be used in a constant expression
main.cpp:530:47: note: declared here
  530 | bool unpackFile(const char *lfsDir, lfs_info *littlefsFile, const char *destPath) {
      |                                               ^
1 error generated.
make: *** [&lt;builtin&gt;: main.o] Error 1</pre>
</details>
</details>

--- a/pkgs/by-name/mk/mklittlefs/package.nix
+++ b/pkgs/by-name/mk/mklittlefs/package.nix
@@ -1,0 +1,77 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  versionCheckHook,
+  nix-update-script,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "mklittlefs";
+  version = "4.1.0";
+
+  src = fetchFromGitHub {
+    owner = "earlephilhower";
+    repo = "mklittlefs";
+    tag = finalAttrs.version;
+    fetchSubmodules = true;
+    hash = "sha256-qCL5EG5HyUjObaRReptuNqMKKxOnyP8ZQpOKdLV4F80=";
+  };
+
+  postPatch = ''
+    substituteInPlace Makefile \
+      --replace-fail '$(shell git describe --tag)' '${finalAttrs.version}' \
+      --replace-fail '$(shell git -C littlefs describe --tags || echo "unknown")' '2.11.1'
+
+      patchShebangs run_tests.sh
+      patchShebangs tests/test_create
+  '';
+
+  makeFlags = [
+    "BUILD_CONFIG_NAME=-nixos"
+  ];
+
+  env = {
+    CPPFLAGS =
+      "-DLFS_NAME_MAX=255"
+      + lib.optionalString stdenv.hostPlatform.isDarwin " -Wno-error=vla-cxx-extension";
+  };
+
+  enableParallelBuilding = true;
+
+  doCheck = true;
+  checkPhase = ''
+     runHook preCheck
+
+    ./run_tests.sh tests
+
+    runHook postCheck
+  '';
+
+  installPhase = ''
+    runHook preInstall
+
+    install -Dm755 mklittlefs -t $out/bin
+
+    runHook postInstall
+  '';
+
+  nativeInstallCheckInputs = [
+    versionCheckHook
+  ];
+  versionCheckProgramArg = "--version";
+  doInstallCheck = true;
+
+  passthru = {
+    updateScript = nix-update-script { };
+  };
+
+  meta = {
+    description = "Tool to build and unpack littlefs images";
+    homepage = "https://github.com/earlephilhower/mklittlefs";
+    changelog = "https://github.com/earlephilhower/mklittlefs/releases/tag/v${finalAttrs.version}";
+    license = lib.licenses.mit;
+    maintainers = with lib.maintainers; [ liberodark ];
+    mainProgram = "mklittlefs";
+  };
+})

